### DISCUSSION TITLE
Company roles - Custom roles addition

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#89] [Q&A] [answered]
Hi guys,
I love the new roles management that we can do on a company and on the projects.
I have a question about that, is there a way to add other custom roles?

We need a role with the following permissions:
- open branches/tags and commit from console
- view and operate on all `non production` environments (pods restart, deploy and so on)
- view the `production` environments, without permissions to operate (just read logs and see pods, but without the permission to deploy, restart pods and so on)

With the following roles we can reach this behavior for a user, but we need to manually change permissions on all project, or:
- give the `developer` role to the user
- for each project of a company, raise the user role to `mantainer` for `non production` environments only

It would be useful to have a role with this permissions, to be able to choose it at company level.

Thanks

**Comment by albertotessarotto** [comment, marked as answer]
Thanks for the issue. Actually, we are not supporting custom roles, but your request is a very valid use case.
In general, the criticality you have encountered is not only related to the many projects within a company; another factor may be if you have to do this for many users of this company.
It might be useful to bring group logic into the console, so that you can easily assign a role to a set of people, or insert a cross-functional team within a new project.
What do you think? We will try to see internally if we can improve with a proper role your reported use case. Thanks

**Reply by frengooo** [reply]
Hi @albertotessarotto, thanks for your reply.

> In general, the criticality you have encountered is not only related to the many projects within a company; another factor may be if you have to do this for many users of this company.

Yes, in my case I have lot of projects and I need to do that for many users.

> It might be useful to bring group logic into the console

You mean at company level or at instance level? Anyway I agree, handle the groups from the console would be a nice improvement.

> We will try to see internally if we can improve with a proper role your reported use case.

Thanks, let me know please.

**Reply by frengooo** [reply]
Hi @albertotessarotto, any news about this thread?

**Reply by albertotessarotto** [reply]
At the moment, we postpone the topic of simplified management with groups because we have more stringent priorities in the short-term roadmap. Personally, I would avoid thinking of a custom role exclusively with such a specific logic, as the management of this use case may be simplified in the near future. Anyway, I prefer to ask @davidebianchi and @fredmaggiowski for their opinion as well

**Reply by davidebianchi** [reply]
We should think about groups, since grouping users can simplify this configuration, and I would address this request with the group feature.
The custom role can bring some difficulties in the user experience, but sure in the future we will add also the custom roles.

Do you have some short term priority needs @silversoul93?

**Reply by fredmaggiowski** [reply]
I 100% agree with you @davidebianchi

**Reply by frengooo** [reply]
Hi guys, I don't think it's a priority task, but another client asks me for information about this feature.
There isn't a related task on the Roadmap, I'll create it.

**Reply by albertotessarotto** [reply]
great idea to have it already mapped out in the roadmap, I would have done it myself, so thank you
